kratos/python_scripts/sympy_fe_utilities.py

```diff
@@ -1,466 +1,522 @@
-from sympy import *
+import re
+import sympy
 
-def DefineMatrix( name, m,n ):
-    """ This method defines a symbolic matrix
+def DefineMatrix(name, m, n):
+    """
+    This method defines a symbolic matrix.
 
     Keyword arguments:
-    name -- Name of variables.
-    m -- Number of rows.
-    n -- Number of columns.
+    - name -- Name of variables.
+    - m -- Number of rows.
+    - n -- Number of columns.
     """
-    return Matrix( m,n, lambda i,j: var(name+'_%d_%d' % (i,j)) )
+    return sympy.Matrix(m, n, lambda i, j: sympy.var("{name}_{i}_{j}".format(name=name, i=i, j=j)))
 
-def DefineSymmetricMatrix( name, m,n ):
-    """ This method defines a symbolic symmetric matrix
+def DefineSymmetricMatrix(name, m, n):
+    """
+    This method defines a symbolic symmetric matrix.
 
     Keyword arguments:
-    name -- Name of variables.
-    m -- Number of rows.
-    n -- Number of columns.
+    - name -- Name of variables.
+    - m -- Number of rows.
+    - n -- Number of columns.
     """
-    tmp = DefineMatrix(name,m,n)
-
-    # Impose symm
-    for i in range(0,tmp.shape[0]):
-        for j in range(i+1,tmp.shape[1]):
-            tmp[j,i] = tmp[i,j]
-
-    return tmp
+    return sympy.Matrix(m, n, lambda i, j:
+        sympy.var("{name}_{i}_{j}".format(name=name, i=min(i,j), j=max(i,j))))
```

Comment:
Are you sure about this?

Reply:
Yep. With a 4x7 matrix I get:
```
	A_0_0	A_0_1	A_0_2	A_0_3	A_0_4	A_0_5	A_0_6
	A_0_1	A_1_1	A_1_2	A_1_3	A_1_4	A_1_5	A_1_6
	A_0_2	A_1_2	A_2_2	A_2_3	A_2_4	A_2_5	A_2_6
	A_0_3	A_1_3	A_2_3	A_3_3	A_3_4	A_3_5	A_3_6
```
I added a unit test right now to make sure.

Reply:
Very good.